QtSharp/ProcessHelper.cs

```diff
@@ -11,26 +11,42 @@ public static string Run(string path, string args, out string error, bool readOu
             {
                 using (Process process = new Process())
                 {
+                    Console.WriteLine("Run: " + path);
+                    Console.WriteLine("Args: " + args);
                     process.StartInfo.FileName = path;
                     process.StartInfo.Arguments = args;
                     process.StartInfo.UseShellExecute = false;
                     process.StartInfo.RedirectStandardOutput = true;
                     process.StartInfo.RedirectStandardError = true;
-                    process.Start();
-                    if (waitForExit)
-                    {
-                        process.WaitForExit();
-                    }
-                    while (readOutputByLines && !process.StandardOutput.EndOfStream)
+
+                    var reterror = "";
+                    var retout = "";
+                    process.OutputDataReceived += (sender, outargs) => {
+                        if (retout != "" && outargs.Data != "") retout += "\r\n";
+                        retout += outargs.Data;
+                        Console.WriteLine("stdout: {0}", retout);
+                    };
+                    process.ErrorDataReceived += (sender, errargs) =>
                     {
-                        Console.WriteLine(process.StandardOutput.ReadLine());
-                    }
-                    error = process.StandardError.ReadToEnd();
+                        if (reterror != "" && errargs.Data != "") reterror += "\r\n";
+                        reterror += errargs.Data;
+                        Console.WriteLine("stderr: {0}", reterror);
+                    };
+
+                    process.Start();
+                    process.BeginOutputReadLine();
+                    process.BeginErrorReadLine();
+
+                    //if (waitForExit) process.WaitForExit();
```

Comment:
If waitForExit is no longer necessary, please delete the parameter and this commented line.